src/Adapter/MSTest.TestAdapter/Execution/TestClassInfo.cs

```diff
@@ -302,6 +303,7 @@ public void RunClassInitialize(TestContext testContext)
                     finally
                     {
                         IsClassInitializeExecuted = true;
+                        (testContext as ITestContext).ClearDiagnosticMessages();
```

Comment:
You are doing a `as` cast which means that it's possible the value returns `null` which would cause a `NullReferenceException` when calling `ClearDiagnosticMessages`. It's ok to play safe by doing `as` cast but then you should handle the null case. Besides, we could probably add a `Debug.Assert` so that we get some warning if this assumption suddenly becomes invalid.

```suggestion
                        Debug.Assert(testContext is ITestContext);
                        (testContext as ITestContext)?.ClearDiagnosticMessages();
```